manifests/server/config.pp

```diff
@@ -141,7 +141,7 @@
     value => $port,
   }
 
-  if ($password_encryption) and ($version >= 10){
+  if ($password_encryption) and ($version.scanf('%i')[0] >= 10){
```

Comment:
You could make use of the [versioncmp](https://puppet.com/docs/puppet/latest/function.html#versioncmp) function, like this:
```
if ($password_encryption) and (versioncmp($version, "10") >= 0)
```
I've run the acceptance and unit tests over this and it seems to be OK.

Reply:
Also spotted another implementation of this in a pending PR: https://github.com/puppetlabs/puppetlabs-concat/pull/627/files#diff-07b462a4b4082856b785d62dee4f7468R42

Reply:
@cmccrisken-puppet I prefer this fix method actually - it's less opaque than mine. I will test it out.  Thanks!

Reply:
I've tested the alternate method and like the results.  I've rebased the commit history to a single commit.  I'm good with the fix if you all are.